src/rust/mangadventure/template/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+							1 => push_sort_param!(params, asc, "views"),
+							2 => push_sort_param!(params, asc, "latest_upload"),
+							3 => push_sort_param!(params, asc, "chapter_count"),
+							_ => continue,
+						}
+					}
+				}
+				FilterType::Genre => {
+					let mut id = encode_uri_component(filter.name.to_lowercase());
+					match filter.value.as_int().unwrap_or(-1) {
+						0 => {
+							id.insert(0, '-');
+							categories.push(id);
+						}
+						1 => categories.push(id),
+						_ => continue,
+					}
+				}
+				_ => continue,
+			}
+		}
+
+		if !categories.is_empty() {
+			params.push_encoded(String::from("categories"), Some(categories.join(",")));
+		}
+
+		self.get_manga_page_result(format!(
+			"{}/api/v2/series?page={}&{}",
+			self.base_url, page, params
+		))
+	}
+
+	pub fn get_manga_listing(&self, listing: Listing, page: i32) -> Result<MangaPageResult> {
+		let mut url = format!("{}/api/v2/series?page={}", self.base_url, page);
+
+		match listing.name.as_str() {
+			"Most Viewed" => url.push_str("&sort=-views"),
+			"Latest Updates" => url.push_str("&sort=-latest_upload"),
+			_ => url.push_str("&sort=title"),
+		}
+
+		self.get_manga_page_result(url)
+	}
+
+	pub fn get_manga_details(&self, id: String) -> Result<Manga> {
+		let json = json_request(format!("{}/api/v2/series/{}", self.base_url, id));
+		let mut url = get_value!(json, url, as_string).read();
+		url.insert_str(0, self.base_url);
+		let id = get_value!(json, slug, as_string).read();
+		let title = get_value!(json, title, as_string).read();
+		let cover = get_value!(json, cover, as_string).read();
+		let description = get_value!(json, description, as_string).read();
+		let author = get_array_as_vec!(json, authors).join(", ");
+		let artist = get_array_as_vec!(json, artists).join(", ");
+		let categories = get_array_as_vec!(json, categories);
+		// TODO: add licensed status when supported
+		let status = if get_value!(json, completed, as_bool) {
+			MangaStatus::Completed
+		} else {
+			MangaStatus::Ongoing
+		};
+		let nsfw = if vec_intersects(&NSFW_CATEGORIES, &categories) {
+			MangaContentRating::Nsfw
+		} else if vec_intersects(&SUGGESTIVE_CATEGORIES, &categories) {
+			MangaContentRating::Suggestive
+		} else {
+			MangaContentRating::Safe
+		};
+
+		Ok(Manga {
+			id,
+			url,
+			title,
+			cover,
+			author,
+			artist,
+			description,
+			categories,
+			status,
+			nsfw,
+			viewer: MangaViewer::Rtl,
+		})
+	}
+
+	pub fn get_chapter_list(&self, id: String) -> Result<Vec<Chapter>> {
+		let json = json_request(format!(
+			"{}/api/v2/chapters?series={}&date_format=timestamp",
+			self.base_url, id
+		));
+		let results = get_value!(json, results, as_array);
+		let mut chapters = vec_from_array::<Chapter>(&results);
+
+		for result in results {
+			let obj = result.as_object().unwrap();
+			let mut url = get_value!(obj, url, as_string).read();
+			url.insert_str(0, self.base_url);
+			let title = get_value!(obj, full_title, as_string).read();
```

Comment:
Wouldn't it make more sense to use the title key? full_title includes the volume and chapter number which Aidoku already displays, so it's duplicate information.
```suggestion
			let title = get_value!(obj, title, as_string).read();
```